py/runai_model_streamer/runai_model_streamer/file_streamer/file_streamer.py

```diff
@@ -12,14 +12,35 @@
 from runai_model_streamer.file_streamer.requests_iterator import (
     RequestsIterator,
 )
+
+from runai_model_streamer.s3_utils.s3_utils import (
+    S3Credentials,
+    is_s3_path,
+    is_gs_path,
+    set_gs_environment_variables,
+    convert_gs_path,
+)
+
 import humanize
+import importlib
 
+s3_module = importlib.util.find_spec("runai_model_streamer_s3") if importlib.util.find_spec("runai_model_streamer_s3") else None
+s3_credentials_module_name = "runai_model_streamer_s3.credentials.credentials"
+if s3_module is not None and importlib.util.find_spec(s3_credentials_module_name):
+    s3_credentials_module = importlib.import_module(s3_credentials_module_name)
+else:
+    s3_credentials_module = None
```

Comment:
I would put all this bulk of code into `runai_model_streamer/s3_utils/s3_utils.py` as long with the `handle_object_store` function, so it will keep FileStreamer clean from this logic and it would be all inside a `s3_util` package